yt_dlp/extractor/epoch.py

```diff
@@ -35,21 +49,59 @@ class EpochIE(InfoExtractor):
                 'id': '9489f994-2a20-4812-b233-ac0e5c345632',
                 'ext': 'mp4',
                 'title': 'Dick Morris Discusses His Book ‘The Return: Trump’s Big 2024 Comeback’',
+                'description': 'md5:ef6f45dd925b5d23e2a862487b42030e',
+                'thumbnail': r're:https://.*\.jpg',
             }
         },
+        {
+            'note': 'Georestricted to US',
+            'url': 'https://www.theepochtimes.com/silence-patton-documentary_4849547.html',
+            'info_dict': {
+                'id': '4849547',
+                'ext': 'mp4',
+                'title': 'Silence Patton | Documentary',
+                'description': 'Why was General Patton silenced during his service in World War II?',
+                'thumbnail': r're:https://.*\.jpg',
+            },
+            'params': {'skip_download': True},
+            'expected_warnings': [
+                'This film is only available in the United States because of territorial licensing.'
+            ],
+        },
     ]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
+        # get_element_by_class() fails on some websites, let's do it the dirty way ...
+        raw_description = clean_html(self._search_regex(
+            r'(?s)<div +class="article_content" *>(.+?)</div>', webpage, 'description', default=''))
+        description_lines_iter = iter(raw_description.splitlines())
+        description = next(description_lines_iter, None)
+        if re.match(r'This film is only available ', description or ''):
+            self.report_warning(description, video_id)
+            description = next(description_lines_iter, None)
+
+        player = extract_attributes(get_element_html_by_class('player-container', webpage) or '<>')
+        youmaker_video_id = try_get(
+            player, lambda x: re.fullmatch(r'player-([0-9a-f-]+)', x['data-id']).group(1))
+
+        if youmaker_video_id:
+            video_id = youmaker_video_id
+            manifest_url = f'http://vs1.youmaker.com/assets/{youmaker_video_id}/playlist.m3u8'
+        elif player.get('data-source', '').endswith('.m3u8'):
+            manifest_url = player['data-source']
+        else:
+            raise UnsupportedError(url)
```

Comment:
is this expected? - i.e. should user make bug report for this? If not expected, raise an ExtractorError instead

Reply:
not expected